app/models/course/condition.rb

```diff
@@ -20,7 +20,9 @@ class << self
     # @return [Object] acts_as_conditionals objects belonging to the given course
     def conditionals_for(course)
       dependent_class_to_condition_class_mapping.keys.map do |conditional_name|
-        conditional_name.constantize.where(course_id: course)
+        if conditional_name.constantize.include? ActiveRecord::Base::ConditionalInstanceMethods
```

Comment:
is_a? seems to be working on the instance of the class only. Will continue to use include? for this part unless you have other alternative.